cpp/src/slice2cs/Gen.cpp

```diff
@@ -874,27 +874,21 @@ Slice::CsVisitor::writeParameterDocComments(const DocComment& comment, const Par
 void
 Slice::CsVisitor::moduleStart(const ModulePtr& p)
 {
```

Comment:
We already validate that `cs:namespace` is only allowed on top-level modules.
If it wasn't we threw the metadata out.

So, checking for `isTopLevel` here is redundant and unnecessary.

Reply:
I don't understand this code. Say we have no metadata whatsoever - we don't generate a C# "namespace Xxx" here?

Reply:
This function seems to be about generating an additional namespace declaration if I read the code correctly. 

```
void
Slice::CsVisitor::moduleStart(const ModulePtr& p)
{
    if (p->isTopLevel())
    {
        string ns = getNamespacePrefix(p);
        if (!ns.empty())
        {
            _out << sp;
            _out << nl << "namespace " << ns;
            _out << sb;
        }
    }
}
```

This function and the corresponding `moduleEnd` are called the visitors.

```
bool
Slice::Gen::TypesVisitor::visitModuleStart(const ModulePtr& p)
{
    moduleStart(p);
    _out << sp;
    _out << nl << "namespace " << p->mappedName();
    _out << sb;

    return true;
}


```

Reply:
They should be renamed `modulePrefixStart` and `modulePrefixEnd` as they're only about the `cs:namespace` metadata.

Reply:
Renamed them as suggested.
I agree, the previous names were confusing.

It took me some time to realize that these are just random helper functions named `moduleStart`,
not the _actual_ code-gen functions `visitModuleStart`. Probably the same as @bernardnormier.